needy/cache.py

```diff
@@ -0,0 +1,29 @@
+
+class Cache:
+    def __init__(self, configuration={}):
+        self.__configuration = configuration
+
+    @staticmethod
+    def type():
+        '''type of cache'''
+        raise NotImplementedError('type')
+
+    def description(self):
+        '''uri, path, or other configuration description'''
+        raise NotImplementedError('description')
+
+    def configuration(self):
+        '''configuration dictionary to serialize to configuration files'''
+        return self.__configuration
+
+    def archive(self, source, key):
+        '''make files and folders at source retrievable with the given key'''
+        raise NotImplementedError('archive')
+
+    def unarchive(self, destination, key):
```

Comment:
Well, I'm not suggesting you don't tarball these directories... but I think refactoring the methods like that reduces implementation code and increases functionality.